streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -960,126 +960,65 @@ KTable<K, V> toTable(final Named named,
                          final Materialized<K, V, KeyValueStore<Bytes, byte[]>> materialized);
 
     /**
-     * Group the records of this {@code KStream} on a new key that is selected using the provided {@link KeyValueMapper}
-     * and default serializers and deserializers.
+     * Group the records by their current key into a {@link KGroupedStream} while preserving the original values.
      * {@link KGroupedStream} can be further grouped with other streams to form a {@link CogroupedKStream}.
-     * Grouping a stream on the record key is required before an aggregation operator can be applied to the data
-     * (cf. {@link KGroupedStream}).
-     * The {@link KeyValueMapper} selects a new key (which may or may not be of the same type) while preserving the
-     * original values.
-     * If the new record key is {@code null} the record will not be included in the resulting {@link KGroupedStream}
-     * <p>
-     * Because a new key is selected, an internal repartitioning topic may need to be created in Kafka if a
-     * later operator depends on the newly selected key.
-     * This topic will be named "${applicationId}-&lt;name&gt;-repartition", where "applicationId" is user-specified in
-     * {@link StreamsConfig} via parameter {@link StreamsConfig#APPLICATION_ID_CONFIG APPLICATION_ID_CONFIG},
+     * (Co-)Grouping a stream on the record key is required before a windowing or aggregation operator can be applied
+     * to the data (cf. {@link KGroupedStream}).
+     * If the grouping key is {@code null} the record will be dropped.
```

Comment:
`grouping key` is not mentioned before. In the context of this method, it's just the key, right?

Reply:
That's correct. I did phrase it this way, because `groupBy(KeyValueMapper)` just links to this method...

I hope it's still clear also in this context, as we start with `Group the records by their current key`, what implies that the current key is the grouping key. At least I hope it's not confusing?

Reply:
I think you can guess it. Without the context of the other method, it just looks a bit weird. The reader will think "why is he not just saying key", but will probably still get it.

Reply:
Pushed an update. Will merge later.